src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/v2/NodeService.kt

```diff
@@ -183,125 +188,169 @@ class NodeService(
+              else null
+            authorizationCode =
+              if (transactionOutcome == ClosePaymentOutcome.OK)
+                authCompleted.transactionAuthorizationCompletedData.authorizationCode
+              else null
+            creationDate = authCompleted.creationDate.toOffsetDateTime()
+            timestampOperation =
+              authCompleted.transactionAuthorizationCompletedData.timestampOperation
+            psp =
+              PspDto().apply {
+                idPsp = authCompleted.transactionAuthorizationRequestData.pspId
+                idChannel = authCompleted.transactionAuthorizationRequestData.pspChannelCode
+                businessName = authCompleted.transactionAuthorizationRequestData.pspBusinessName
+                brokerName = authCompleted.transactionAuthorizationRequestData.brokerName
+                pspOnUs = authCompleted.transactionAuthorizationRequestData.isPspOnUs
+              }
           }
-        else null
-      transactionDetails =
-        TransactionDetailsDto().apply {
-          transaction =
-            TransactionDto().apply {
-              this.transactionId = transactionId.value()
-              transactionStatus = getTransactionDetailsStatus(authCompleted)
-              paymentGateway = authCompleted.transactionAuthorizationRequestData.paymentGateway.name
-              this.fee = feeEuroCents
-              this.amount = amountEuroCents
-              grandTotal = totalAmountEuroCents
-              rrn = authCompleted.transactionAuthorizationCompletedData.rrn
-              errorCode =
-                if (transactionOutcome == ClosePaymentRequestV2Dto.OutcomeEnum.KO)
-                  getAuthorizationErrorCode(
+        info =
+          InfoDto().apply {
+            type = authCompleted.transactionAuthorizationRequestData.paymentTypeCode
+            brandLogo =
+              authCompleted.transactionAuthorizationRequestData
+                .transactionGatewayAuthorizationRequestedData
+                .logo
+                .toString()
+            brand =
+              when (authRequestedData) {
+                is NpgTransactionGatewayAuthorizationRequestedData -> authRequestedData.brand
+                is PgsTransactionGatewayAuthorizationRequestedData ->
+                  authRequestedData.brand?.toString()
+                is RedirectTransactionGatewayAuthorizationRequestedData ->
+                  authRequestedData.paymentMethodType.toString()
+                else -> null
+              }
+            paymentMethodName = authCompleted.transactionAuthorizationRequestData.paymentMethodName
+            clientId = authCompleted.transactionActivatedData.clientId.name
+          }
+        user = UserDto().apply { type = UserDto.TypeEnum.GUEST }
+      }
+
+    return when (authRequestedData.type!!) {
+      TransactionGatewayAuthorizationRequestedData.AuthorizationDataType.PGS,
+      TransactionGatewayAuthorizationRequestedData.AuthorizationDataType.NPG ->
```

Comment:
I would add here a comment for NPG case to handle close payment for paypal apm methods (as a reminder for the future) and wallet

Reply:
Added in 620ec21